src/backend/opencl/platform.cpp

```diff
@@ -906,22 +908,36 @@ DeviceManager::DeviceManager()
                 break;
             }
         }
-
         if (!default_device_set) {
             printf("WARNING: AF_OPENCL_DEFAULT_DEVICE_TYPE=%s is not available\n",
                     deviceENV.c_str());
             printf("Using default device as 0\n");
         }
     }
 
+    try {
+        // If forgeManager instantiation fails,
+        // ignore and proceed
+        fgMngr.reset(new graphics::ForgeManager());
```

Comment:
I have thought about that but it required more changes to the way DependencyModule class was used earlier. Since, this check was happening only once during DeviceManager, didn't expect it to have any performance impact. 

I have addressed all other feedback already and pushing this change in different commit. If everything checks out, we can squash-merge.